lib/galaxy/tools/actions/upload_common.py

```diff
@@ -45,7 +45,13 @@
 
 def validate_datatype_extension(datatypes_registry, ext):
     if ext and ext not in ("auto", "data") and not datatypes_registry.get_datatype_by_extension(ext):
+        # also allow putting in data ontology ... might want to model this as a separate input
+        if ":" in ext:
+            ext_by_ontology = datatypes_registry.get_datatype_ext_by_format_ontology(ext)
```

Comment:
I think that method is only in the CWL branch atm ?

Generally speaking, this was a quick hack to allow specifying datatypes by the edam ontology, I don't know if we should perhaps be more explicit here and have a separate input to determine this ?